### ISSUE TITLE
Read Me

### ISSUE THREAD

**Issue by katulu-anne** [hypha-space/hypha#185]
## Hypha

<p align="center">
  <img src="docs/assets/hypha-architecture-overview.svg"
       alt="Hypha coordinating distributed workers and a data node from a single training script">
</p>

Hypha is a self-managing system for distributed machine learning – think **“Kubernetes for AI” without the Kubernetes cluster**. It discovers and connects heterogeneous compute (CPUs, GPUs, accelerators) over unreliable networks and runs training and inference jobs across them.

Under the hood, Hypha uses **libp2p** for decentralized networking and **DiLoCo-style training** to cut communication by orders of magnitude compared to classic data-parallel setups.

Instead of hand-wiring SSH tunnels, tmux sessions, and brittle launch scripts, you describe the job and let Hypha handle:

- **Resource discovery & orchestration** across machines and networks  
- **Low-communication distributed training** using DiLoCo (≈500× less traffic than naive data-parallel)  
- **Secure, fault-tolerant execution** with no single control-plane node  

---

### Is Hypha for you?

Hypha is for teams that:

- Have **multiple machines with GPUs/CPUs** (on-prem servers, homelab boxes, and/or some cloud VMs)  
- Currently juggle training with **SSH, tmux, and ad-hoc scripts**  
- Want **distributed training** without running **Kubernetes or Slurm**  
- Live with **imperfect networks** (VPN, mixed subnets, flaky links) but still want something more robust than “hope the script doesn’t die”

---

### Turn your heterogeneous GPU cluster into one logical execution fabric

You run a small research group or startup with:

- an **8×GPU server** in a rack,  
- a couple of **smaller GPU machines** under people’s desks,  
- and a few **cloud VMs** you spin up when experiment queues explode.

Today you either:

- ignore half the hardware, or  
- manually juggle runs with SSH, ad-hoc scripts, and “who’s using what?” spreadsheets.

With Hypha you:

1. **Install Hypha** on each machine (no Kubernetes required).  
2. **Run one gateway and one scheduler**, then start **workers** next to each GPU and a **data node** next to the dataset.  
3. **Submit your distributed training job once**; Hypha discovers the workers, coordinates data loading, and runs a DiLoCo training job across them – even if some links are slow or flaky.  

If that picture feels familiar, stop juggling SSH and scripts and let Hypha turn your hardware mess into one logical training system.

---

## Quickstart

Want to see Hypha do something real in under an hour?  
👉 Follow the Quickstart guide and walk through this setup step by step:

https://hypha-space.org/quickstart/

---

### Choose your path

- 🧪 **Just want to try it quickly?**  
  Start with the [Quickstart](https://hypha-space.org/quickstart/) to bring up a small Hypha cluster and run a distributed training job.

- 🧩 **Want to integrate Hypha into your own code / workflows?**  
  Check the usage and API docs (TODO: link to docs) for how to submit jobs and wire Hypha into your training loops.

- 🔧 **Want to hack on Hypha itself?**  
  See [CONTRIBUTING.md](CONTRIBUTING.md) for development setup, code structure, and how to propose changes.
